packages/libraries/cli/src/base-command.ts

```diff
@@ -142,17 +145,41 @@ export default abstract class extends Command {
   }
 
   registryApi(registry: string, token: string) {
-    const client = new GraphQLClient(registry, {
-      headers: {
-        Accept: 'application/json',
-        'User-Agent': `hive-cli/${this.config.version}`,
-        Authorization: `Bearer ${token}`,
-        'graphql-client-name': 'Hive CLI',
-        'graphql-client-version': this.config.version,
-      },
-    });
+    const requestHeaders = {
+      Accept: 'application/json',
+      'User-Agent': `hive-cli/${this.config.version}`,
+      Authorization: `Bearer ${token}`,
+      'graphql-client-name': 'Hive CLI',
+      'graphql-client-version': this.config.version,
+    };
+
+    return {
+      async request<TData = any, TVariables = Record<string, any>>(operation: TypedDocumentNode<TData, TVariables>, variables: TVariables) {
```

Comment:
Here we can use the function signature as mentioned in https://github.com/kamilkisiela/graphql-hive/issues/1703#issuecomment-1588634237

https://github.com/dotansimha/graphql-code-generator/blob/2139f1db5704aa34a0cd89342736bf67ec556da3/examples/yoga-tests/src/yoga.spec.ts#L6-L23

The benefit of this one is that `TVariables` will be an optional parameter, based on whether variables are defined or not.

Reply:
much better. thanks